src/js/components/FocusedContainer.js

```diff
@@ -12,13 +12,92 @@ export const FocusedContainer = ({
 }) => {
   const [bodyOverflowStyle, setBodyOverflowStyle] = useState('');
   const ref = useRef(null);
+  const preNodeRef = useRef(null);
+  const postNodeRef = useRef(null);
+
+  const { contextValue, withinGrommet } = useRoots();
+  const { roots: contextRoots } = contextValue;
 
-  const roots = useContext(RootsContext);
-  const [nextRoots, setNextRoots] = useState(roots);
   useEffect(() => {
-    // make sure value of null is not added to array
-    if (ref.current) setNextRoots([...roots, ref.current]);
-  }, [roots]);
+    const container = ref.current;
+    const roots = contextRoots.current;
+
+    const handleTrapFocus = (e) => {
+      if (!hidden && trapFocus && container) {
+        const focusableElements = container.querySelectorAll(
+          `button:not([tabindex="-1"]), [href]:not([tabindex="-1"]), 
+     input:not([tabindex="-1"]), select:not([tabindex="-1"]), 
+     textarea:not([tabindex="-1"]), [tabindex]:not([tabindex="-1"])`,
+        );
+
+        if (focusableElements.length === 0) return;
+
+        const firstElement = focusableElements[0];
+        const lastElement = focusableElements[focusableElements.length - 1];
+
+        // only perform focus if this is the most recently opened drop
+        if (roots[roots.length - 1] === container) {
```

Comment:
would it be worth moving this check up between 26 and 27 so it avoids doing the query when it's not the top container?

Reply:
Done.